### PR TITLE
feat(android): per-conversation notification settings (XEP-0492)

### PR DESCRIPTION
**Pull request by icepuma**
## Summary

Per-conversation notification settings (mute) for the Android app, wired end-to-end over XEP-0492 (Chat Notification Settings) on top of the existing core implementation in `waddle-xmpp-client`: new UniFFI verbs, a session store hydrated on every fresh stream, a bell sheet in the conversation top bar, mute glyphs in the channel/DM lists, and XEP-0492 gating in the notification policy.

## Rust (`server/`)

- **Core (`waddle-xmpp-client/src/xep/xep0492/notify.rs`)**: lifted the read-side surface out of the WASM crate into core — `NotifySettingsSurface`, `surface_bookmark_notify` (scans ALL `<notify/>` siblings for the fallback mode + rich-payload opt-in), `surface_dm_notify`. The WASM crate's `surface_bookmark` / `surface_dm_bookmark` now delegate to it, so both binding crates share one implementation. 6 new core tests in the XEP-0492 suite.
- **FFI (`waddle-xmpp-client-ffi/src/notify_settings.rs`, new)**:
  - Types: `WaddleNotifyMode { Always, OnMention, Never }`, `WaddleBookmarkItem`, `WaddleDmBookmarkItem`, `WaddleSetRoomNotificationModeOutcome { Ok{item}, NodeConfigMismatch, Error }`, `WaddleSetDmNotificationModeOutcome { Ok{item}, Removed{jid}, NodeConfigMismatch, Error }` — mirroring the core outcome shapes.
  - Verbs: `fetch_user_bookmarks()` / `fetch_dm_bookmarks()` (`item-not-found` → empty list, WASM parity, 30s IQ timeout) and `set_room_notification_mode(room_jid, mode, name, rich_payload_opt_in)` / `set_dm_notification_mode(dm_jid, mode, rich_payload_opt_in)` delegating to core `notification_settings` with `ClientHandle` as the `CommandDriver` behind a timeout-bounding adapter (30s per IQ, matching the fetch verbs). Domain-only JIDs and unparsable JIDs reject early as `WaddleError::InvalidJid`; bookmark names normalize empty/whitespace to `None` (WASM parity).
  - `notify_settings_tests.rs` (24 tests) mirrors the WASM boundary cases: item-not-found → empty, opt-in read, fallback read across fallbackless notify siblings, DM domain-only item skip, outcome mapping, IqTimeout → Timeout collapse, not-connected/invalid-JID typed failures.
- **Bindings**: committed Kotlin (`apps/android/.../ffi/waddle_xmpp_client.kt`) and Swift (`apps/apple/Waddle/RustClient/Generated/*`) bindings regenerated; both drift-check scripts pass.

## Android (`apps/android`)

- **`store/NotifySettingsStore.kt`** (+ `ConversationKind`, `defaultNotifyMode`, `NotifySettingsEntry`): JID-keyed entries, `modeFor(jid, kind)` resolving `null` → XEP-0492 §3 defaults (*"always" for direct chats and private group chats, "on-mention" for public group chats*), generation-guarded commit-on-success `hydrate()` merging both carriers (a `clear()` or local set racing the fetch discards the stale snapshot), re-entrancy guard, `clear()` on logout. Registered in `SessionStores`, re-exported on `XmppSessionManager`. **§3 decision**: the topology has no public/private room discriminator, so every MUC resolves as `PRIVATE_GROUP` (default `always`) — exact web parity with `chat/src/lib/notify-settings.ts`.
- **Hydration**: `SessionCatchup.onSessionReady` hydrates on every **fresh** stream (web `runSessionBootstrap` parity — a XEP-0198 resume preserves state; no PEP `+notify` subscription yet, deferred like the web).
- **Verbs**: `ConversationVerbs.setRoomNotificationMode` / `setDmNotificationMode` (+facade passthroughs) returning the new sealed `NotifySettingsResult { Ok, NodeConfigMismatch, NotConnected, Rejected }`; the current rich-payload opt-in is repeated from the store so a mode change never clobbers it; the store reconciles from `Ok`/`Removed` outcomes without a refetch.
- **UI**: bell `IconButton` in `ConversationScreen`'s TopAppBar (parent conversations only; `NotificationsOff` glyph when muted) opening `NotifySettingsSheet` — a `ModalBottomSheet` radio list (All messages / Mentions only / Muted with the web's label+hint copy), current mode pre-selected from the store. `NodeConfigMismatch` gets its own carrier-aware snackbar (room vs DM copy, web parity); other failures reuse the shared action-failure snackbars. Mute glyphs on muted rows in `HomeScreen` (body list + drawer badge) and `DmListScreen`.
- **Notifier gating**: `NotificationPolicy` takes a `notifyModeFor(jid, isGroupchat)` snapshot function (kept pure/testable): `NEVER` suppresses outright, `ON_MENTION` requires `messageMentionsBareJid` (broadcast or self URI), `ALWAYS` keeps current behavior. Wired through `MessageNotifier` from `AppGraph` reading the store.
- **Fakes**: `FakeWaddleClient` implements the four new interface members with canned lists/outcomes + recorded calls; the DM default mirrors the sparse carrier (`always` without opt-in → `Removed`).

## Tests

- Rust: 6 new core XEP-0492 surface tests, 24 new FFI tests (`cargo test -p waddle-xmpp-client -p waddle-xmpp-client-ffi -p waddle-xmpp-client-wasm` green).
- Android JVM: `NotifySettingsStoreTest` (9: §3 defaults, hydrate merge, clear/set races, re-entrancy, reconciles), verb tests in `XmppSessionManagerVerbsTest` (5), hydrate sequencing in `XmppSessionManagerCatchupTest` (4: fresh hydrates both carriers, overrides + defaults, resume skips refetch, logout clears), `NotificationPolicyTest` gating matrix (6 new).
- Android instrumentation (GMD atdApi34): `NotifySettingsSheetTest` — opens the bell sheet, asserts the §3 default pre-selection, picks Muted, asserts the fake client recorded the XEP-0492 set call and the store updated, reopens and asserts the override pre-selects.

## Verification

- `cargo test -p waddle-xmpp-client -p waddle-xmpp-client-ffi -p waddle-xmpp-client-wasm` — green
- `cargo clippy --all-targets -p waddle-xmpp-client -p waddle-xmpp-client-ffi -- -D warnings`, `cargo fmt --all -- --check` — clean
- `scripts/check-android-ffi-bindings.sh` + `server/scripts/check-xmpp-client-ffi-bindings.sh` — no drift
- `./gradlew detekt testDebugUnitTest :app:lintDebug :app:assembleDebug` — green
- `scripts/build-android-rust.sh --debug` + `./gradlew --no-configuration-cache :app:atdApi34DebugAndroidTest` — green
- Wire shapes verified against `xeps/xep-0492.xml` + `xeps/xep-0402.xml` (no wire changes in this PR — the carriers were already implemented and tested in core)

## Review

Two adversarial review rounds (XMPP/Rust + Android/Kotlin personas). Round-1 findings fixed: the set verbs now share the fetch verbs' 30s IQ timeout via a `CommandDriver` adapter; the store's hydrate commit is serialized with the set-verb reconciles behind a monitor (closes a check-to-assign race window); the bell gates on the view-model's thread-ness instead of the `onOpenThread` capability; the `IqTimeout → Timeout` collapse is pinned by a test. Round 2: no remaining actionable issues from either reviewer.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/android/app/src/main/kotlin/social/waddle/android/feature/channel/ChannelViewModel.kt [hunk 1]
```diff
@@ -34,6 +37,9 @@ class ChannelViewModel(
     pinnedIds = sessionManager.pinStore.pinnedIds(roomJid),
     mentionCandidates = sessionManager.presenceStore.occupants
         .map { rooms -> mentionCandidatesOf(rooms[roomJid].orEmpty()) },
```
[the remark this answers is not in the document]
Comment (icepuma) [reply apps/android/app/src/main/kotlin/social/waddle/android/feature/channel/ChannelViewModel.kt:39]: Declined — the quoted sentence does not exist in XEP-0492. §3 in full says: "In the absence of any notification settings for a given conversation, 'always' SHOULD be assumed for direct chats and private group chats, and on-mention for public group chats." (xeps/xep-0492.xml line 136). There is no 'cannot be determined → on-mention' clause; the XEP is silent on undetermined types. Resolving undetermined MUCs as private-group/always is a legitimate interpretation and matches the web client (chat/src/lib/notify-settings.ts). When the topology gains a public/private discriminator, both clients should switch together.
